test/integration/scheduler/queue_test.go

```diff
@@ -327,6 +327,36 @@ func TestCoreResourceEnqueue(t *testing.T) {
 			// because QHint of TaintToleration would decide to ignore a Pod update.
 			enableSchedulingQueueHint: []bool{true},
 		},
+		{
+			name: "Pod rejected by the PreFilter of NodeAffinity plugin and Filter of NodeResourcesFit is requeued when a Node is created and turned to ready",
+			initialNodes: []*v1.Node{
+				st.MakeNode().Name("fake-node").Label("node", "fake-node").Capacity(map[v1.ResourceName]string{v1.ResourceCPU: "2"}).Obj(),
+				st.MakeNode().Name("fake-node2").Label("node", "fake-node2").Label("zone", "zone1").Capacity(map[v1.ResourceName]string{v1.ResourceCPU: "2"}).Obj(),
+			},
+			pods: []*v1.Pod{
+				// - Pod1 will be rejected by the NodeAffinity plugin and NodeResourcesFit plugin.
+				st.MakePod().Label("unscheduled", "plugins").Name("pod1").NodeAffinityIn("metadata.name", []string{"fake-node"}).Req(map[v1.ResourceName]string{v1.ResourceCPU: "4"}).Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) error {
+				// Trigger a NodeDeleted event.
+				// It will not requeue pod1 to activeQ
```

Comment:
```suggestion
				// It will not requeue pod1 to activeQ
				// because both NodeAffinity and NodeResourceFit don't register Node/delete event. 
```